src/main/java/org/sqlite/SQLiteJDBCLoader.java

```diff
@@ -326,20 +327,16 @@ private static void loadSQLiteNativeLibrary() throws Exception {
         }
 
         // Load the os-dependent library from the jar file
-        String packagePath = SQLiteJDBCLoader.class.getPackage().getName().replaceAll("\\.", "/");
-        sqliteNativeLibraryPath =
-                String.format(
-                        "/%s/native/%s", packagePath, OSInfo.getNativeLibFolderPathForCurrentOS());
-        boolean hasNativeLib = hasResource(sqliteNativeLibraryPath + "/" + sqliteNativeLibraryName);
-
-        if (!hasNativeLib) {
-            if (OSInfo.getOSName().equals("Mac")) {
-                // Fix for openjdk7 for Mac
-                String altName = "libsqlitejdbc.jnilib";
-                if (hasResource(sqliteNativeLibraryPath + "/" + altName)) {
-                    sqliteNativeLibraryName = altName;
-                    hasNativeLib = true;
-                }
+        sqliteNativeLibraryPath = LibraryLoaderUtil.getNativeLibResourcePath();
+        boolean hasNativeLib =
+                LibraryLoaderUtil.hasNativeLib(sqliteNativeLibraryPath, sqliteNativeLibraryName);
+
+        if (!hasNativeLib && OSInfo.getOSName().equals("Mac")) {
+            // Fix for openjdk7 for Mac
+            String altName = "libsqlitejdbc.jnilib";
+            if (LibraryLoaderUtil.hasNativeLib(sqliteNativeLibraryPath, altName)) {
+                sqliteNativeLibraryName = altName;
+                hasNativeLib = true;
             }
```

Comment:
Docs indicate this is only required for JDK7, minimal version is JDK8 so can this be removed?

Reply:
yes, well spotted, we can remove that